explorer/client/src/components/displaybox/DisplayBox.tsx

```diff
@@ -75,29 +93,57 @@ function DisplayBox({ display }: { display: string }) {
         );
     } else {
         return (
-            <div className={styles['display-container']} id="displayContainer">
-                {!hasDisplayLoaded && (
-                    <div className={styles.imagebox} id="pleaseWaitImage">
-                        Image Loading...
+            <>
+                {hasClickedImage && (
+                    <div
+                        className={styles.modalcontainer}
+                        onClick={handleImageClick}
+                    >
+                        <div className={styles.modal}>
+                            <img
+                                id="loadedImage"
```

Comment:
Is this ID needed? My concern here would be that having an ID would limit component re-use, given IDs need to be globally unique.

Reply:
The IDs are used exclusively for finding elements under the e2e tests rather than for stylization.